jbmc/src/java_bytecode/java_bytecode_instrument.cpp

```diff
@@ -592,6 +593,27 @@ void java_bytecode_instrument_symbol(
   instrument(to_code(symbol.value));
 }
 
+/// Instruments the start function with an assertion that checks whether
+/// an exception has escaped the entry point
+/// \param init_code: the code block to which the assertion is appended
+/// \param exc_symbol: the top-level exception symbol
+/// \param source_location: the source location to attach to the assertion
+void java_bytecode_instrument_uncaught_exceptions(
+  codet &init_code,
+  const symbolt &exc_symbol,
+  const source_locationt &source_location)
+{
+  // check that there is no uncaught exception
+  code_assertt assert_no_exception;
+  assert_no_exception.assertion() = equal_exprt(
+    exc_symbol.symbol_expr(),
+    null_pointer_exprt(to_pointer_type(exc_symbol.type)));
+  source_locationt assert_location = source_location;
```

Comment:
Is there actual value in this temporary?

Reply:
It's shorter than `assert_no_exception.add_source_location().set_comment("no uncaught exception");`